pkg/weather-client/types.go

```diff
@@ -0,0 +1,98 @@
+package client
+
+import "time"
+
+type Elevation struct {
+	UnitCode string  `json:"unitCode"`
+	Value    float64 `json:"value"`
+}
+
+type Dewpoint struct {
+	UnitCode string  `json:"unitCode"`
+	Value    float64 `json:"value"`
+}
+
+type ProbabilityOfPrecipitation struct {
+	UnitCode string `json:"unitCode"`
+	Value    int    `json:"value"`
+}
+
+type RelativeHumidity struct {
+	UnitCode string `json:"unitCode"`
+	Value    int    `json:"value"`
+}
+
+type Period struct {
+	Dewpoint                   Dewpoint                   `json:"dewpoint"`
+	EndTime                    time.Time                  `json:"endTime"`
+	Icon                       string                     `json:"icon"`
+	IsDaytime                  bool                       `json:"isDaytime"`
+	Name                       string                     `json:"name"`
+	Number                     int                        `json:"number"`
+	ProbabilityOfPrecipitation ProbabilityOfPrecipitation `json:"probabilityOfPrecipitation"`
+	RelativeHumidity           RelativeHumidity           `json:"relativeHumidity"`
+	ShortForecast              string                     `json:"shortForecast"`
+	StartTime                  time.Time                  `json:"startTime"`
+	Temperature                int                        `json:"temperature"`
+	TemperatureTrend           interface{}                `json:"temperatureTrend"`
+	TemperatureUnit            string                     `json:"temperatureUnit"`
+	WindDirection              string                     `json:"windDirection"`
+	WindSpeed                  string                     `json:"windSpeed"`
+}
+
+type Properties struct {
+	Elevation         Elevation `json:"elevation"`
+	ForecastGenerator string    `json:"forecastGenerator"`
+	GeneratedAt       time.Time `json:"generatedAt"`
+	Periods           []Period  `json:"periods"`
+	Units             string    `json:"units"`
+	UpdateTime        time.Time `json:"updateTime"`
+	Updated           time.Time `json:"updated"`
+	ValidTimes        string    `json:"validTimes"`
+}
+
+type ForecastResponse struct {
+	Properties Properties `json:"properties"`
+	Type       string     `json:"type"`
+}
+
+type PointResponse struct {
```

Comment:
Are you generating these response types?

I recommend not embedding struct types unless there's a really solid reason. There are a few benefits to not embedding:

1. Reusability of "sub-structs". 
    * For example, if `Geometry` ends up getting used in more than 1 response, it doesn't need to be re-declared/maintained.
2. Docs
    * When structs aren't embedded we get to add docs (comments) to them describing what they are/for/etc. This enables IDEs to provide this context when folks are accessing them.

See [Unembed ForcastResponse struct](https://github.com/dsauerbrun/cragcast/pull/5/commits/632d4947032fba18611ec2f3f7215514a220b066) for en example of decoupling these.

Reply:
yeah, I've been using gojsonstruct as a stopgap until I work on #3 

This is pure laziness on my part since I wanted to work on other stuff that was less boring.